internal/directory/okta/okta.go

```diff
@@ -133,6 +152,9 @@ func (p *Provider) getGroups(ctx context.Context) ([]*directory.Group, error) {
 		RawQuery: fmt.Sprintf("limit=%d", p.cfg.batchSize),
 	}).String()
 	for groupURL != "" {
+		if err := p.limiter.Wait(ctx); err != nil {
+			return nil, fmt.Errorf("reached limit: %w", err)
```

Comment:
I think this error message isn't quite right. Wait will error out if the context is cancelled or the deadline exceeded, but doesn't error out if the limit is exceeded.

Reply:
Also can we move this call to the `apiGet` so we only have to do it once?

Reply:
It can return error in case we reached the burst size.

Yes, moving to `apiGet` is right 👍